src/test/java/guitests/BoardAutoCreatorTest.java

```diff
@@ -72,4 +72,43 @@ public void milestoneBoardAutoCreationTest() {
         assertEquals(panelInfos.get(4).getPanelName(), "Next Next Next Milestone");
     }
 
+
+    @Test
+    public void workAllocationBoardAutoCreationTest() {
+        assertEquals(panelControl.getNumberOfSavedBoards(), 0);
+
+        click("Boards");
+        push(KeyCode.DOWN); // Save
+        push(KeyCode.DOWN); // Save As
+        push(KeyCode.DOWN); // Open
+        push(KeyCode.DOWN); // Delete
+        push(KeyCode.DOWN); // Auto-create
+        push(KeyCode.RIGHT); // Open Auto-create, Milestones
+        push(KeyCode.DOWN); // Work Allocation
+        push(KeyCode.ENTER);
```

Comment:
Yes, this has been the practice for all GUI menu tests and I think it will
break the menu changes. I will try to make such function.
On Oct 27, 2015 12:43 PM, "Damith C. Rajapakse" <notifications@github.com>
wrote:

> In src/test/java/guitests/BoardAutoCreatorTest.java
> <https://github.com/HubTurbo/HubTurbo/pull/1093#discussion_r43084010>:
>
> > @@ -72,4 +72,43 @@ public void milestoneBoardAutoCreationTest() {
> >          assertEquals(panelInfos.get(4).getPanelName(), "Next Next Next Milestone");
> >      }
> >
> > +
> > +    @Test
> > +    public void workAllocationBoardAutoCreationTest() {
> > +        assertEquals(panelControl.getNumberOfSavedBoards(), 0);
> > +
> > +        click("Boards");
> > +        push(KeyCode.DOWN); // Save
> > +        push(KeyCode.DOWN); // Save As
> > +        push(KeyCode.DOWN); // Open
> > +        push(KeyCode.DOWN); // Delete
> > +        push(KeyCode.DOWN); // Auto-create
> > +        push(KeyCode.RIGHT); // Open Auto-create, Milestones
> > +        push(KeyCode.DOWN); // Work Allocation
> > +        push(KeyCode.ENTER);
>
> This is going to break when the menu changes in future. May be write a
> function that can find an item and click it irrespective of menu content?
> e.g. clickMenu(String... items) and call it clickMenu("Boards","Auto-create","Work
> distribution") Note the use of variable length argument lists.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1093/files#r43084010>.
>